components/Hero.tsx

```diff
@@ -1,27 +1,157 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styled from "@emotion/styled";
+import Image from "next/image";
+import { useQueryClient } from "react-query";
 import { useRouter } from "next/router";
+import { matchSorter } from "match-sorter";
+
+import {
+  useCombobox,
+  UseComboboxState,
+  UseComboboxStateChangeOptions,
+} from "downshift";
+
 import { MaxWidthWrapper } from "./Wrapper";
-import { SearchBar } from "./SearchBar";
-import { QUERIES } from "../utils/constants";
+import SearchIcon from "../public/icons/search.svg";
+import {
+  QUERIES,
+  formatContentfulUrl,
+  CATEGORIES_PLACEHOLDERS,
+} from "../utils";
+import { Emp } from "../utils/umaApi";
 
 type HeroProps = {
   topAction?: React.ReactNode;
 };
+
 export const Hero: React.FC<HeroProps> = ({ children, topAction = null }) => {
   const router = useRouter();
+  const queryClient = useQueryClient();
+  const searchMap = queryClient.getQueryData<Emp[]>("all synths");
+  const [matches, setMatches] = useState<Emp[]>([]);
+
+  const onSelectAction = (selectedSynth?: Emp | null) => {
+    if (!selectedSynth) {
+      return;
+    }
+    router.push(`/${selectedSynth.address}`);
+  };
+  // we use this to hook into downshift state changes and route to a new page when an item is selected.
+  const stateReducer = (
+    _state: UseComboboxState<Emp>,
+    actionAndChanges: UseComboboxStateChangeOptions<Emp>
+  ) => {
+    const { type, changes } = actionAndChanges;
+    switch (type) {
+      case useCombobox.stateChangeTypes.ItemClick:
+      case useCombobox.stateChangeTypes.InputKeyDownEnter:
+      case useCombobox.stateChangeTypes.InputBlur:
+        onSelectAction(changes.selectedItem);
+        return changes;
+      default:
+        return changes;
+    }
+  };
+  const {
+    isOpen,
+    getMenuProps,
+    highlightedIndex,
+    getItemProps,
+    getInputProps,
+    getComboboxProps,
+  } = useCombobox<Emp>({
+    items: matches,
+    itemToString: (item) => item?.address ?? "",
+    stateReducer,
+    onInputValueChange: ({ inputValue }) => {
+      if (!inputValue) {
+        setMatches([]);
+        return;
+      }
+      if (!searchMap) {
+        return;
+      }
+      setMatches(
```

Comment:
i tested searching on the site, was suprised it was pretty fast as i typed, but if this becomes a performance issue we might want to use lodash.debounce or something to defer searching till typing stops

Reply:
Agree! I had the debounced the function initially but couldn't see any big difference, so I've removed it to make the code simpler